packages/renderer/src/lib/kube/pods/KubernetesTerminalBrowser.spec.ts

```diff
@@ -0,0 +1,154 @@
+/**********************************************************************
+ * Copyright (C) 2024-2025 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import '@testing-library/jest-dom/vitest';
+
+import { render, screen } from '@testing-library/svelte';
+import userEvent from '@testing-library/user-event';
+import { beforeAll, beforeEach, expect, test, vi } from 'vitest';
+
+import { podsInfos } from '/@/stores/pods';
+
+import { terminalService } from '../../pod/KubernetesTerminalService';
+import KubernetesTerminalBrowser from './KubernetesTerminalBrowser.svelte';
+import type { PodUI } from './PodUI';
+
+vi.mock('@xterm/xterm', () => {
+  return {
+    Terminal: vi
+      .fn()
+      .mockReturnValue({ loadAddon: vi.fn(), open: vi.fn(), write: vi.fn(), dispose: vi.fn(), onData: vi.fn() }),
+  };
+});
```

Comment:
```suggestion
vi.mock('@xterm/xterm');
```

You are calling `vi.clearAllMock()` so this mock is reset before the test begin

Reply:
Changing this causes a failure:
```
TypeError: Cannot read properties of undefined (reading 'serialize')

	in KubernetesTerminalBrowser.svelte

 ❯ src/lib/pod/KubernetesTerminal.svelte:56:36
     54| 
     55| onDestroy(() => {
     56|   terminalContent = serializeAddon.serialize();
```

Reply:
werid, probably an issue with the way we mock, let's keep it for now